app/database/server/models/post_metadata.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Relation} from '@nozbe/watermelondb';
+import Model, {Associations} from '@nozbe/watermelondb/Model';
+import {field, immutableRelation, json} from '@nozbe/watermelondb/decorators';
+
+import {MM_TABLES} from '@constants/database';
+import Post from '@typings/database/post';
+
+const {POST, POST_METADATA} = MM_TABLES.SERVER;
+
+/**
+ * PostMetadata provides additional information on a POST
+ */
+export default class PostMetadata extends Model {
+    /** table (entity name) : PostMetadata */
+    static table = POST_METADATA;
+
+    /** associations : Describes every relationship to this entity. */
+    static associations: Associations = {
+
+        /** A POST can have multiple POST_METADATA.(relationship is 1:N)*/
+        [POST]: {type: 'belongs_to', key: 'post_id'},
+    };
+
+    /** post_id : The foreign key of the parent POST model */
+    @field('post_id') postId!: string;
+
+    /** type : The type will work in tandem with the value present in the field 'data'.  One 'type' for each kind of 'data' */
+    @field('type') type!: string;
```

Comment:
check https://github.com/mattermost/mattermost-mobile/blob/master/app/mm-redux/types/posts.ts#L14 for possible values

Reply:
Corrected